practice/tic-tac-toe/list_board.py

```diff
@@ -0,0 +1,137 @@
+        >>> a._rows = [[' ', 'X', 'O'], [' ', ' ', ' '], [' ', ' ', ' ']]
+        >>> a.place_token(0, 0, 'O')
+        >>> a
+        ListTTTBoard([['O', 'X', 'O'], [' ', ' ', ' '], [' ', ' ', ' ']])
+        >>> a = ListTTTBoard()
+        >>> a._rows = [[' ', 'X', 'O'], [' ', ' ', ' '], [' ', ' ', ' ']]
+        >>> a.place_token(0, 1, 'O')
+        >>> a
+        ListTTTBoard([[' ', 'X', 'O'], ['O', ' ', ' '], [' ', ' ', ' ']])
+        """
+        (self._rows[y])[x] = token
+
+    def calc_winner(self):
+        """Determines what token string has won or returns None if no one has
+
+        >>> a = ListTTTBoard()
+        >>> a._rows = [[' ', ' ', ' '], [' ', ' ', ' '], [' ', ' ', ' ']]
+        >>> a.calc_winner() == None
+        True
+
+        >>> a = ListTTTBoard()
+        >>> a._rows = [
+        ... ['X', 'X', 'O'],
+        ... ['O', 'O', 'X'],
+        ... ['X', 'O', 'X']
+        ... ]
+        >>> a.calc_winner() != None
+        False
+
+        >>> a = ListTTTBoard()
+        >>> a._rows = [
+        ... [' ', ' ', ' '],
+        ... [' ', ' ', ' '],
+        ... ['X', 'X', 'X']
+        ... ]
+        >>> a.calc_winner()
+        'X'
+
+        >>> a = ListTTTBoard()
+        >>> a._rows = [
+        ... [' ', ' ', 'O'],
+        ... [' ', ' ', 'O'],
+        ... ['X', 'X', 'O']
+        ... ]
+        >>> a.calc_winner()
+        'O'
+
+        >>> a = ListTTTBoard()
+        >>> a._rows = [
+        ... ['X', ' ', ' '],
+        ... ['O', 'X', 'O'],
+        ... ['O', 'X', 'X']
+        ... ]
+        >>> a.calc_winner()
+        'X'
+        """
+        win_lines = [
+            [(self._rows[1])[0], (self._rows[1])[1], (self._rows[1])[2]],
+            [(self._rows[0])[1], (self._rows[1])[1], (self._rows[2])[1]],
+            [(self._rows[0])[0], (self._rows[1])[1], (self._rows[2])[2]],
+            [(self._rows[0])[2], (self._rows[1])[1], (self._rows[2])[0]],
+            [(self._rows[0])[0], (self._rows[0])[1], (self._rows[0])[2]],
+            [(self._rows[0])[0], (self._rows[1])[0], (self._rows[2])[0]],
+            [(self._rows[2])[0], (self._rows[2])[1], (self._rows[2])[2]],
+            [(self._rows[0])[2], (self._rows[1])[2], (self._rows[0])[2]]
+        ]
+        if ['X', 'X', 'X'] in win_lines:
+            return 'X'
+        elif ['O', 'O', 'O'] in win_lines:
+            return 'O'
+        else:
+            return None
+
+    def __str__(self):
+        r"""Returns a pretty-printed picture of the board.
+
+        >>> a = ListTTTBoard()
+        >>> a._rows = [
+        ... ['X', 'X', 'O'],
+        ... ['O', 'O', 'X'],
+        ... ['X', ' ', 'X']
+        ... ]
+        >>> a.__str__()
+        'X|X|O\nO|O|X\nX| |X\n'
+        """
+        return self._rows[0][0] + '|' + self._rows[0][1] + '|' + self._rows[0][2] + '\n' +\
+            self._rows[1][0] + '|' + self._rows[1][1] + '|' + self._rows[1][2] + '\n' +\
+            self._rows[2][0] + '|' + self._rows[2][1] + '|' + self._rows[2][2] + '\n'
```

Comment:
There is a really nice parallel between `str.join()` and what the goal of this function is. Try using it instead of manually interpolating.